### ISSUE TITLE
Problem compiling

### ISSUE THREAD

**Issue by cs-victor-nascimento** [potatosalad/erlang-libdecaf#1]
Hey @potatosalad! I was trying to test autodiscovery of libdecaf in JOSE (with Joken) but I can't compile.

I get this error:

``` sh
/home/victornascimento/dev/elixir/joken/deps/libdecaf/c_src/libdecaf_nif.h:152:2: error: unknown type name ‘uint8_t’
  uint8_t   version;
```

I am on Ubuntu 15.10 64 bits. If you need more info I will gladly fetch it.

Thanks!

**Comment by potatosalad** [comment]
Hey @cs-victor-nascimento!

The main [libdecaf](https://sourceforge.net/projects/ed448goldilocks/) library is still being finalized and made more portable (right now I've only tested it on OS X and SmartOS).  An "almost final" version of the [EdDSA draft](https://tools.ietf.org/html/draft-irtf-cfrg-eddsa-04) was released yesterday, so the EdDSA functions in libdecaf will probably be finalized soon and it will actually be released.  [Mike Hamburg](https://github.com/bitwiseshiftleft) is the author of the library, so I'm not sure about an exact release date.

That said, the `uint8_t` error is probably just due to a missing `#include <stdint.h>` in the `c_src/libdecaf_nif.h` file.  I'll see if I can sync the latest changes from libdecaf and get that added, too.

**Comment by cs-victor-nascimento** [comment]
Thanks for clarifying! I tested libsodium and it worked great. I was wondering how faster this was going to be.

Do you think it is best to let this issue open?

**Comment by potatosalad** [comment]
@cs-victor-nascimento I just published [version 0.0.2 to hex.pm](https://hex.pm/packages/libdecaf/0.0.2) and I finally got the builds passing on [travis with clang 3.7](https://travis-ci.org/potatosalad/erlang-libdecaf).

On Linux, you may need `gcc >= 4.7`, `g++ >= 4.7`, and/or `clang >= 3.6` for this to build properly.  If you're still having problems, try building with environment variables like `CC=gcc-4.7 CXX=g++-4.7 ARCHFLAGS=` prefixing `mix` or `make` or however you're installing this.

Let me know if that works and we'll close this issue.

**Comment by cs-victor-nascimento** [comment]
Nice! Just checked it and it works perfectly. The boost in speed is also unbelievable though expected.

Looking forward to your work on libsodium for implementing timeslice reductions in those NIFs too!

Thanks once again!
